### DISCUSSION TITLE
How to change the default mapping of `<CR>` on `dap-repl`?

### DISCUSSION THREAD

**Discussion by nyngwang** [mfussenegger/nvim-dap#835] [Q&A] [answered]
If I press `<CR>` on `dap-repl` buffer then I will get a warning like this:

```
DAP No action possible on: Type 'dlv help' for list of commands.
```

then I found that it is a result of the underlying command:

```lua
lua require('dap.ui').trigger_actions({ mode = 'first' })
```

which is set by default in both 

https://github.com/mfussenegger/nvim-dap/blob/700a3c7d6fbe5b07bee74e9952b456120d355c47/lua/dap/ui/widgets.lua#L10-L11

and 

https://github.com/mfussenegger/nvim-dap/blob/a9f21a209323a8554da19bebd132936e7d48126d/lua/dap/repl.lua#L32

Since I already have a normal map for `<CR>`, I would like to know how to change/disable these mappings by default.

---

**UPDATE**: I came across an issue where you shared [an example usage of `dap.listeners.after` by your dotfiles](https://github.com/mfussenegger/dotfiles/blob/e7abb9a13f8fb3075704ed703dd973ecf3502cc3/vim/.config/nvim/lua/me/dap.lua#L64-L75). While that seems the most related to my problem here, I still don't know how to override the default `<CR>` keymap given this info.

**Comment by mfussenegger** [comment, marked as answer]
With https://github.com/mfussenegger/nvim-dap/commit/43695d2cb7992436c5533cbfd58ed8d53d2c8be3 you should be able to setup a `FileType` autocmd for `dap-repl` to override/remove the keymap.

```lua
vim.api.nvim_create_autocmd('FileType', {
  pattern = 'dap-repl',
  callback = function(args)
    vim.keymap.del('n', '<CR>', { buffer = args.buf })
  end
})
```

**Reply by nyngwang** [reply]
Thank you for supporting this!

**Reply by nyngwang** [reply]
@mfussenegger It seems that this `FileType` autocmd can work only after first calling `dap.continue()`. It doesn't work after the buffer got **updated** after calling either `dap.step_into/out/over()`.

**update**: Just resolved this problem by using `WinEnter` instead. I guess that the `vim.bo.filetype` is **not** always set upon the `FileType` event, or maybe the default mapping `<CR>` is added back by some other autocmd.

```lua
vim.api.nvim_create_autocmd({ 'WinEnter' }, {
  pattern = '*',
  callback = function ()
    if vim.bo.filetype == 'dap-repl' then
      pcall(vim.keymap.del, 'n', '<CR>', { buffer = vim.api.nvim_get_current_buf() })
    end
  end
})
```
